packages/flutter_tools/lib/src/commands/channel.dart

```diff
@@ -123,14 +123,26 @@ class ChannelCommand extends FlutterCommand {
 
   Future<void> _switchChannel(String branchName) async {
     globals.printStatus("Switching to flutter channel '$branchName'...");
-    if (!kOfficialChannels.contains(branchName)) {
+    if (kObsoleteBranches.containsKey(branchName)) {
+      final String alternative = kObsoleteBranches[branchName]!;
+      globals.printStatus("This channel is obsolete. Consider switching to the '$alternative' channel instead.");
+    } else if (!kOfficialChannels.contains(branchName)) {
       globals.printStatus('This is not an official channel. For a list of available channels, try "flutter channel".');
     }
     await _checkout(branchName);
     globals.printStatus("Successfully switched to flutter channel '$branchName'.");
     globals.printStatus("To ensure that you're on the latest build from this channel, run 'flutter upgrade'");
   }
 
+  static Future<void> upgradeChannel(FlutterVersion currentVersion) async {
+    final String channel = currentVersion.channel;
+    if (kObsoleteBranches.containsKey(channel)) {
+      final String alternative = kObsoleteBranches[channel]!;
+      globals.printStatus("Transitioning from '$channel' to '$alternative'...");
+      return _checkout(alternative);
```

Comment:
Question: should we prompt the user first before migrating them to beta?

Reply:
nah